js/bartender.js

```diff
@@ -1,86 +1,192 @@
+                        $('.user-role').show();
+                        return role;
+                    } else {
+                  }
+                });
+            }); // end barkeep tasks
+
+        } else {
+            // get customer preferences
+            $('.customer').show();
+
+            $('.wine-flavor').hide();
+            $('.wine-weight').hide();
+
+            var BlendPreference = new CustPreferences([]);
+
+            $('.wine-color').prepend("<p>Do you prefer red wine or white wine?</p>");
+            $('.wine-color').show();
+            $('.wine-color').click( function(e) {
+                e.preventDefault();
+                wineColor = $('input[name=wineColor]:checked').val();
+
+                BlendPreference.addPreference(wineColor);
+                $('.wine-color').hide();
+
+                // get wine flavor preference
+                $('.wine-flavor').prepend("<p>And do you like your " + wineColor + " wine sweet or dry?</p>");
+                $('.wine-flavor').show();
+                $('.wine-flavor').click( function(e) {
+                    e.preventDefault();
+                    wineFlavor = $('input[name=wineFlavor]:checked').val();
+
+                    BlendPreference.addPreference(wineFlavor);
+                    $('.wine-flavor').hide();
+
+                    // get wine weight - heavy or light - preference
+                    $('.wine-weight').prepend("<p>Do you like a full-bodied " + wineFlavor + " " + wineColor + " " + " wine or a lighter blend?</p>");
+                    $('.wine-weight').show();
+                    $('.wine-weight').click( function(e) {
+                        e.preventDefault();
+                        wineWeight = $('input[name=wineWeight]:checked').val();
+
+                        BlendPreference.addPreference(wineWeight);
+                        $('.wine-weight').hide();
+
+                        // bartender has to make a wine blend
+                        var barkeep = new Barkeep('Lynne', 'makeWineBlend');
+
+                        // get new wine blend name and mix
+                        wineName = barkeep.newBlend(BlendPreference);
+
+                        // clear input text and radio values
+                        $('input[name=wineName]').val('');
+                        $('input[name=wineColor]').prop('checked', false);
+                        $('input[name=wineFlavor]').prop('checked', false);
+                        $('input[name=wineWeight]').prop('checked', false);
+                        $('input[name=restock]').prop('checked', false);
+
+                        // deliver new wine blend
+                        $('.deliver-blend').show();
+                        $('.deliver-blend').prepend("<p><br>" + barkeep.name + " created " + wineName + " blend for you!</p>");
+
+                        });  // end get wine weight preference
+
+                  }); // end get wine flavor preference
+
```

Comment:
I wouldn't have the click event listeners nested inside each other. The classes to which we register for click events are already all in the html when the page loads. Also, I don't see any use of variables from other click event handler scopes. This is making the code look very complex with all the indentation and nesting. I'd take all listeners out and put them all inside the document.ready with the same scope. Every handler function then simply updates the UI as needed, adds to our preferences object, etc.